TheFoodieNetwork/app/src/main/java/com/fbu/thefoodienetwork/adapters/LocationAdapter.java

```diff
@@ -0,0 +1,77 @@
+package com.fbu.thefoodienetwork.adapters;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.fbu.thefoodienetwork.R;
+import com.fbu.thefoodienetwork.models.Location;
+
+import java.util.List;
+
+public class LocationAdapter extends RecyclerView.Adapter<LocationAdapter.ViewHolder> {
+    private static final String TAG = "LocationAdapter";
+    private Context context;
+    private List<Location> locationList;
+    private OnClickLocationListener onClickLocationListener;
+
+    public LocationAdapter(Context context, List<Location> locationList) {
+        this.context = context;
+        this.locationList = locationList;
+
+        try {
+            this.onClickLocationListener = ((OnClickLocationListener) context);
+        } catch (ClassCastException e) {
+            throw new ClassCastException("Activity must implement OnClickLocationListener.");
```

Comment:
Since you're already catching the exception here, you can just put `throw e` unless you really want to add a custom exception message.  Another workaround would be to add a `Log.e` above the `throw` statement.